README.md

```diff
@@ -14,22 +14,27 @@ A proposed [WebAssembly System Interface](https://github.com/WebAssembly/WASI) A
 
 ### Phase 4 Advancement Criteria
 
-`wasi-runtime-config` should have at least two implementations (i.e., from service providers, and or cloud providers), and, at the very minimum, pass the testsuite for Windows, Linux, and MacOS.
+`wasi-runtime-config` should have at least two implementations (i.e., from service providers, and or
+cloud providers), and, at the very minimum, pass the testsuite for Windows, Linux, and MacOS.
 
 ## Table of Contents
 
 - [Introduction](#introduction)
 
 ### Introduction
 
-The `wasi-runtime-config` world aim to provide a set of generic interfaces for dynamic configuration management systems. They are often the source of the configuration truth that lives in external services. Configuration values are often polled by the application for
+The `wasi-runtime-config` world aim to provide a set of generic interfaces for providing
+configuration to a component. Configuration values are often polled by the application for
 
-1. toggling on/off feature flags
-2. A/B testing
-3. reading user allow/deny lists
+1. Configuring the runtime behavior of a component. Yes, that sounds generic, but applications do
+   all sorts of crazy things with configuration values! Calling entirely different branches of code,
+   setting upstream URLs or services, configuring the number of threads to use, etc.
+2. toggling on/off feature flags
+3. A/B testing
 
 and many more use cases.
```

Comment:
Is secrets management in-scope?

Reply:
This is an interesting discussion. For now, I'm going to say no. But I think that access to secret values could be through this interface. Implementors (generally hosts) would need to make sure that secret values are injected into the available config